### DISCUSSION TITLE
How to add an app to LXQt Modules?

### DISCUSSION THREAD

**Discussion by stefonarch** [lxqt/lxqt#2127] [Q&A] [answered]
Switching from compton to picom I've noticed that in debian (but not in arch) there is a picom module, half working: it starts and stops picom well, and I can set it to start at session start but at next opening of lxqt-session-conf it's unset again.

![schermata-11-16-17-39](https://user-images.githubusercontent.com/10681413/142028710-4baaeae4-e4e9-4234-b579-324bdc346601.png)

What are the conditions for an app to get registered as module? The only one I know is[ lxqt-appswitcher](https://github.com/zjes/lxqt-appswitcher).

**Comment by tsujan** [comment]
picom as an LXQt module? So weird! Something's wrong.

**Reply by tsujan** [reply]
You may have made an auto-start desktop file with `X-LXQt-Module=true` in it (in `/etc/xdg/autostart/` or `~/.config/autostart/`)

**Reply by stefonarch** [reply]
```
$ cat /etc/xdg/autostart/picom.desktop 
[Desktop Entry]
Type=Application
Name=Picom (X Compositor)
GenericName=X compositor
Comment=A X compositor
TryExec=picom
Exec=picom --dbus

# turned off by default
Hidden=true
OnlyShowIn=LXQt;
X-LXQt-Module=true

```

I thought I didn't make this but uninstalling picom it still emains and it's coming from [here](https://github.com/redtide/picom-conf/blob/picom-conf/autostart/picom.desktop.in). It's also impossible to add it as application in autostart, getting an "file picom.desktop already exists".
I can add it under another name and then changing the name.
But the question remains: why doesn't it work completely as module?

**Reply by stefonarch** [reply]
reply to self: I had also a symlink from some days ago in `~/.config/autostart/picom.conf → /etc/xdg/autostart/picom.conf` , didn't enable "show hidden" first. Without the link the picom as module looks working fine.

**Comment by stefonarch** [comment, marked as answer]
So the answer is: any application which has `X-LXQt-Module=true` in its *.desktop file will be seen under "Modules" and can be started and stopped from there. I used it also for the dropdown qterminal which has no "close" button.

**Reply by tsujan** [reply]
Good to know that you found the answer. I didn't remember the code; was going to find and read it again. Now, I let it be forgotten until the next time :P
